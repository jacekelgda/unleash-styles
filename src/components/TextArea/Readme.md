TextArea

    <TextArea id='foo' label='bar' placeholder='baz' />
