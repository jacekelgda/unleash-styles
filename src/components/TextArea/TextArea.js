import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './TextArea.css'

/**
 * TextArea
 */
class TextArea extends Component {
  render () {
		return (
			<textarea className={styles.TextArea} {...this.props} />
		)
  }
}

TextArea.propTypes = {
}

export default TextArea
