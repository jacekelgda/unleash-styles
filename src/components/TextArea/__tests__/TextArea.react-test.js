/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import TextArea from '../TextArea'

test('TextArea renders correctly', () => {
  const wrapper = shallow(<TextArea id='foo' label='bar' placeholder='baz' />)
  expect(wrapper).toMatchSnapshot()
})
