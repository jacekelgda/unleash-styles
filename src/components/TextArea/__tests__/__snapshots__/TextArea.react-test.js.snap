// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <textarea
    className="TextArea"
    id="foo"
    label="bar"
    placeholder="baz"
/>,
  "nodes": Array [
    <textarea
      className="TextArea"
      id="foo"
      label="bar"
      placeholder="baz"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TextArea
        id="foo"
        label="bar"
        placeholder="baz"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TextArea {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "id": "foo",
          "label": "bar",
          "placeholder": "baz",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <textarea
          className="TextArea"
          id="foo"
          label="bar"
          placeholder="baz"
/>,
        "_debugID": 2,
        "_renderedOutput": <textarea
          className="TextArea"
          id="foo"
          label="bar"
          placeholder="baz"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextArea
    id="foo"
    label="bar"
    placeholder="baz"
/>,
}
`;
